terracotta/drivers/relational_base.py

```diff
@@ -0,0 +1,486 @@
+                self.DATABASE_DRIVER_EXCEPTIONS_TO_CONVERT
+            ):
+                return fun(self, *args, **kwargs)
+        return inner
+    return decorator
+
+
+@contextlib.contextmanager
+def convert_exceptions_context(
+    error_message: str,
+    exceptions_to_convert: Union[Type[Exception], Tuple[Type[Exception], ...]]
+) -> Iterator:
+    try:
+        yield
+    except exceptions_to_convert as exception:
+        raise exceptions.InvalidDatabaseError(error_message) from exception
+
+
+class RelationalDriver(RasterDriver, ABC):
+    SQL_DATABASE_SCHEME: str  # The database flavour, eg mysql, sqlite, etc
+    SQL_DRIVER_TYPE: str  # The actual database driver, eg pymysql, sqlite3, etc
+    SQL_KEY_SIZE: int
+    SQL_TIMEOUT_KEY: str
+
+    FILE_BASED_DATABASE: bool = False
+
+    DATABASE_DRIVER_EXCEPTIONS_TO_CONVERT: Tuple[Type[Exception], ...] = (
+        sqla.exc.OperationalError,
+        sqla.exc.InternalError,
+        sqla.exc.ProgrammingError,
+        sqla.exc.InvalidRequestError,
+    )
+
+    SQLA_REAL = functools.partial(sqla.types.Float, precision=8)
+    SQLA_TEXT = sqla.types.Text
+    SQLA_BLOB = sqla.types.LargeBinary
+    _METADATA_COLUMNS: Tuple[Tuple[str, sqla.types.TypeEngine], ...] = (
+        ('bounds_north', SQLA_REAL),
+        ('bounds_east', SQLA_REAL),
+        ('bounds_south', SQLA_REAL),
+        ('bounds_west', SQLA_REAL),
+        ('convex_hull', SQLA_TEXT),
+        ('valid_percentage', SQLA_REAL),
+        ('min', SQLA_REAL),
+        ('max', SQLA_REAL),
+        ('mean', SQLA_REAL),
+        ('stdev', SQLA_REAL),
+        ('percentiles', SQLA_BLOB),
+        ('metadata', SQLA_TEXT)
+    )
+
+    def __init__(self, path: str) -> None:
+        settings = terracotta.get_settings()
+        db_connection_timeout: int = settings.DB_CONNECTION_TIMEOUT
+
+        assert self.SQL_DRIVER_TYPE is not None
+        self._CONNECTION_PARAMETERS = self._parse_connection_string(path)
+        cp = self._CONNECTION_PARAMETERS
+        resolved_path = self._resolve_path(cp.path[1:])  # Remove the leading '/'
+        connection_string = f'{cp.scheme}+{self.SQL_DRIVER_TYPE}://{cp.netloc}/{resolved_path}'
+
+        self.sqla_engine = sqla.create_engine(
+            connection_string,
+            echo=True,
+            future=True,
+            connect_args={self.SQL_TIMEOUT_KEY: db_connection_timeout}
+        )
+        self.sqla_metadata = sqla.MetaData()
+
+        self._db_keys: Optional[OrderedDict] = None
+
+        self.connection: Connection
+        self.connected: bool = False
+        self.db_version_verified: bool = False
+
+        # use normalized path to make sure username and password don't leak into __repr__
+        qualified_path = self._normalize_path(path)
+        super().__init__(qualified_path)
+
+    @classmethod
+    def _parse_connection_string(cls, connection_string: str) -> urlparse.ParseResult:
+        con_params = urlparse.urlparse(connection_string)
```

Comment:
```suggestion
        if "//" not in connection_string:
            connection_string = f"//{connection_string}"

        con_params = urlparse.urlparse(connection_string)
```

This fixes your failing test and is still clean enough IMO.

Reply:
Ah, neat solution! This required one extra change a few lines below. I just committed that locally, as I couldn't figure out how to edit your suggestion here. Sorry for not giving you the credit in the logs